### PR TITLE
Enhanced bot-updates script to update MWE version

### PR DESCRIPTION
**Pull request by kthoms**
Adds a stage 'MWE_VERSION' that does the various changes for updating
the MWE version in target definitions, POM's, feature.xml and wizard
classes.

The effect can be seen as dry run in the following build (copied bot-updates job for testing)
https://ci.eclipse.org/xtext/job/experimental/job/bot-updates-test/15/console

Note that the replaced values you see there are not properly correct, the values have been chosen for demo purposes.
The input value for param `UPDATE_VALUE` is `1.5.1.M3,2.11.1.M3,https://download.eclipse.org/modeling/emft/mwe/updates/milestones/S201910141054`.
This is the first example where the value contains multiple values as a list. This is done to avoid the need for more parameters.

The first 2 args are replaced where MWE is referenced by its Maven coordinates. That's in the BOM and some pom.xml files.
The URL is replaced in target definitions.

**Note:** In xtext-umbrella we need the OSGi versions of the bundles, which differs from the Maven versions set as input. To avoid entering even more values the replacement values are determined by listing the directory contents of the p2 repository's `plugins/` folder and filtering for a MWE2 bundle.

**Note:** Currently the pom.xml for the domainmodel example would not be updated properly. To ease the replacement it is planned to define a property `<mwe2Version>` like in other POMs, which makes the replacement for the required dependency version easier with sed. Searching for the right <version> tag to update would require to get more from the context, which would make the command harder. If this is OK I plan to provide a separate PR for the domainmodel pom.xml file.

### CODE REVIEW

**Comment by cdietrich** [comment]
i dont think it is a good idea to patch the target platform project this way.
we use the template/rich string there for a reason and its not prepared for that you do here.
you also dont patch the xtext version file

File: releng/jenkins/bot-updates/Jenkinsfile [hunk 1]
```diff
@@ -325,6 +328,69 @@ pipeline {
...
+              sed -i -E 's|(org\\.eclipse\\.emf\\.mwe2\\..*):[^"]*|\\1:${env.MWE2_VERSION}|g' org.eclipse.xtext.dev-bom/build.gradle
+            """
+          }}
+          if (fileExists('xtext-core')) { dir('xtext-core') {
+            sh """
+              for f in \$(find org.eclipse.xtext.tests/testdata/wizard-expectations -name pom.xml -type f); 
+              do
+                sed -i -e 's|<mwe2Version>.*</mwe2Version>|<mwe2Version>${env.MWE2_VERSION}</mwe2Version>|' \$f
+              done
+              sed -i -e 's|https://.*/emft/mwe/updates/[^"]*|${env.MWE_P2_URL}|' org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/TargetPlatformProject.xtend
```
Comment (cdietrich) [inline releng/jenkins/bot-updates/Jenkinsfile:366]: needs rework regaring XtextVersion.java

**Review by cdietrich — CHANGES_REQUESTED** [review]

File: releng/jenkins/bot-updates/Jenkinsfile (hunk 1 above)
Comment (kthoms) [reply releng/jenkins/bot-updates/Jenkinsfile]: Updated. Found a way to replace the next line after match with sed, which was required to change the getters' return values in XtextVersion

https://ci.eclipse.org/xtext/job/experimental/job/bot-updates-test/20/console

```
23:38:17  diff --git a/org.eclipse.xtext.util/src/org/eclipse/xtext/util/XtextVersion.java b/org.eclipse.xtext.util/src/org/eclipse/xtext/util/XtextVersion.java
23:38:17  index ac6c8eb..6a430e2 100644
23:38:17  --- a/org.eclipse.xtext.util/src/org/eclipse/xtext/util/XtextVersion.java
23:38:17  +++ b/org.eclipse.xtext.util/src/org/eclipse/xtext/util/XtextVersion.java
23:38:17  @@ -52,14 +52,14 @@ public class XtextVersion {
23:38:17   	 * @noreference
23:38:17   	 */
23:38:17   	public String getMweVersion() {
23:38:17  -		return "2.12.0.M2";
23:38:17  +		return "2.11.1.M3";
23:38:17   	}
23:38:17   	
23:38:17   	/**
23:38:17   	 * @noreference
23:38:17   	 */
23:38:17   	public String getMweBuildNumber() {
23:38:17  -		return "S202010292115";
23:38:17  +		return "S201910141054";
23:38:17   	}
```

**Comment by kthoms** [comment]
@cdietrich Did the MWE 2.12.0 updates with this job. I guess this change is fine when all version changes are approved.

**Comment by cdietrich** [comment]
@kthoms the version in feature.xml still is false

**Review by NicoPrediger — APPROVED** [review]

**Comment by Hejado** [comment]
Please ignore my approval, made it by mistake because of "no coffee yet"

**Comment by cdietrich** [comment]
move to 2.25

**Comment by cdietrich** [comment]
what is the state here?

**Comment by cdietrich** [comment]
move to 2.26

**Comment by cdietrich** [comment]
with the repo merge this is obsolete for now

**Pull request closed without merging**